cirq-core/cirq/linalg/transformations.py

```diff
@@ -168,6 +169,85 @@ def targeted_left_multiply(
     )  # type: ignore
 
 
+@dataclasses.dataclass
+class _SliceConfig:
+    axis: int
+    source_index: int
+    dest_index: int
```

Comment:
nit: "target_index" as short as "source_index" and doesn't require abbreviation. Also, you actually use "target_index" in the examples below.